src/main/scala/viper/gobra/translator/implementations/translator/StatementsImpl.scala

```diff
@@ -68,6 +68,23 @@ class StatementsImpl extends Statements {
       } yield exhale
     }
 
+    def gatherBreakLabels(node: in.Node, label: Option[String], depthControl : Boolean = true): Vector[String] =
```

Comment:
I'm lacking some intuition what this function is gathering. It seems to collect all `escLabel` of break statements if they are either unlabeled or are labeled `label`. But what's the purpose of `depthControl`?

Reply:
You are right, this is not obvious, I added a javaDoc comment explaining it. Basically, since a break will become a goto to a unique escLabel, we need to gather all these escLabels corresponding to one while loop so we can insert them after the while loop. The escLabels that correspond to a specific loop are the ones that belong to unlabeled breaks directly inside the loop's body and the escLabels of labeled breaks with the same label as the while loop. The argument 'depthControl' is needed to determine if we are directly inside the loop body or we are currently traversing nested while loops so we should disregard unlabeled breaks.